src/Strategy/Gomodules.hs

```diff
@@ -42,5 +44,6 @@ mkProject project =
 
 getDeps :: (Has Exec sig m, Has ReadFS sig m, Has Diagnostics sig m) => GomodulesProject -> m (Graphing Dependency)
 getDeps project =
-  GoList.analyze' (gomodulesDir project)
-    <||> Gomod.analyze' (gomodulesGomod project)
+  context "Gomodules" $
+    context "Dynamic analysis" (GoList.analyze' (gomodulesDir project))
+      <||> context "Static analysis" (Gomod.analyze' (gomodulesGomod project))
```

Comment:
I'm not sure everyone who works on non-scaffolding areas of the code will immediately understand how these multiple, nested `context`s are going to compose.  Some questions they might have immediately:
* Are contexts additive?
* When and how are they dropped and replaced?
* Are they cleared if not replaced by something else?
 
I think the semantics need to be clear to us without trying to make everyone else figure out how haskell is going to handle the effects.  While you need haskell expertise to hack through the scaffolding of this project, I think we should try very hard to make that less true for the business logic, and strategies in particular should be as close to easy as possible.

The solution is dev docs, in my opinion.  I would really like to see how it resolves in frames, i.e. showing a reader the expected printed value, explaining when it moves to the next value, and how the value will be expected to change.

Reply:
Good question -- the semantics of `context` for Diagnostics is that it's additive and scoped:
```hs
context "foo" $ do
  context "bar" $ do
    -- context is `foo -> bar` here
    pure ()
  context "baz" $ do
    -- context is `foo -> baz` here
    pure ()
  -- context is `foo` here
```

here's an example of a deeply nested context stack during gradle discovery:
```
[TASK 6] Gradle > Finding projects > Walking the filetree > Listing gradle projects at '/Users/connor/Desktop/tmp2/grpc-java/' > Running command '/Users/connor/Desktop/tmp2/grpc
-java/gradlew'
```

I'll add more docs to `context` to make that clear. The primary purpose for `context` is to create "traceback"s for thrown errors:

```
[ WARN] ----------
  An error occurred:

      Command execution failed:
          command: gradle
          args: [projects]
          dir: /Users/connor/Desktop/tmp2/grpc-java/
          exit: ExitFailure 1
          stdout:

          stderr:
            gradle: startProcess: runInteractiveProcess: exec: does not exist (No such file or directory)

      Traceback:
        - Running command 'gradle'
        - Listing gradle projects at '/Users/connor/Desktop/tmp2/grpc-java/'
```

..and we're just hijacking it to produce status messages

Reply:
I think the use of context here is slightly confusing, but it does make sense and the resulting errors are very nice looking so I'm happy about that. As long as we document this I am ok with it.